presto-main/src/main/java/io/prestosql/sql/planner/planprinter/IoPlanPrinter.java

```diff
@@ -470,11 +501,33 @@ public Void visitTableScan(TableScanNode node, IoPlanBuilder context)
         {
             TableMetadata tableMetadata = metadata.getTableMetadata(session, node.getTable());
             TupleDomain<ColumnHandle> predicate = metadata.getTableProperties(session, node.getTable()).getPredicate();
+            QualifiedObjectName tableName = new QualifiedObjectName(tableMetadata.getCatalogName().getCatalogName(),
+                    tableMetadata.getTable().getSchemaName(),
+                    tableMetadata.getTable().getTableName());
+            Optional<TableHandle> optTableHandle = metadata.getTableHandle(session, tableName);
+            Long estBytesScanned = null;
+            if (optTableHandle.isPresent()) {
+                TableHandle table = optTableHandle.get();
+                Constraint constraint = new Constraint(predicate);
+                TableStatistics tableStatistics = metadata.getTableStatistics(session, table, constraint);
+                long bytesScanned = 0;
+                // Only include stats for columns that are projected or used in a filter
+                Set<ColumnHandle> predicateColumns = predicate.getDomains().get().keySet();
+                Collection<ColumnHandle> projectedColumns = node.getAssignments().values();
+                for (Map.Entry<ColumnHandle, ColumnStatistics> entry : tableStatistics.getColumnStatistics().entrySet()) {
+                    if (!entry.getValue().getDataSize().isUnknown()
```

Comment:
IMO if any stats for column is unknown we should mark `estBytesScanned` as unknown as well.

Reply:
See comment above.